### PR TITLE
tests/common: migrate auth tests #1

### PR DESCRIPTION
**Pull request by chaochn47**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow. 

Related to https://github.com/etcd-io/etcd/pull/14041

### CODE REVIEW

File: tests/common/auth_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+func TestAuthDisable(t *testing.T) {
+	testRunner.BeforeTest(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1}))
+	defer clus.Close()
+	cc := testutils.MustClient(clus.Client())
+	testutils.ExecuteUntil(ctx, t, func() {
+		assert.NoError(t, cc.Put(ctx, "hoo", "a", config.PutOptions{}))
+		assert.NoErrorf(t, setupAuth(cc, []authRole{testRole}, []authUser{rootUser, testUser}), "failed to enable auth")
```
Comment (ahrtr) [inline tests/common/auth_test.go:48]: Probably you need to use `require.NoErrorf` instead. `assert.NoErrorf` will not terminate the execution of the test case when it fails. Obviously it doesn't make sense to continue to execute the following code once it fails.

File: tests/common/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+		rootAuthClient := testutils.MustClient(clus.Client(WithAuth(rootUserName, rootPassword)))
+		testUserAuthClient := testutils.MustClient(clus.Client(WithAuth(testUserName, testPassword)))
+
+		// test-user doesn't have the permission, it must fail
+		assert.Error(t, testUserAuthClient.Put(ctx, "hoo", "bar", config.PutOptions{}))
+		assert.NoErrorf(t, rootAuthClient.AuthDisable(ctx), "failed to disable auth")
```
Comment (ahrtr) [inline tests/common/auth_test.go:55]: same as above

File: tests/common/auth_test.go (hunk 1 above)
Comment (chaochn47) [reply tests/common/auth_test.go:48]: Make sense. Thanks for the suggestion.

File: tests/common/auth_util.go [hunk 3]
```diff
@@ -20,7 +20,28 @@ import (
...
+	rootUserName = "root"
+	rootRole     = "root"
+	rootPassword = "root"
+	testUserName = "test-user"
+	testRoleName = "test-role"
+	testPassword = "pass"
+)
+
+var (
+	rootUser = authUser{user: rootUserName, pass: rootRole, role: rootPassword}
```
Comment (ahrtr) [inline tests/common/auth_util.go:36]:
```suggestion
	rootUser = authUser{user: rootUserName, pass: rootPassword, role: rootRole}
```

File: tests/common/auth_util.go [hunk 4]
```diff
@@ -20,7 +20,28 @@ import (
 
 	clientv3 "go.etcd.io/etcd/client/v3"
 	"go.etcd.io/etcd/tests/v3/framework/config"
-	intf "go.etcd.io/etcd/tests/v3/framework/interfaces"
+	"go.etcd.io/etcd/tests/v3/framework/interfaces"
+)
+
+const (
+	rootUserName = "root"
+	rootRole     = "root"
```
Comment (ptabor) [inline tests/common/auth_util.go:28]: rootRole = "root"
rootPassword = "rootPassword"

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @chaochn47

**Pull request merged**